runtime/lua/vim/_system.lua

```diff
@@ -211,15 +211,13 @@ end
 --- @param clear_env? boolean
 --- @return string[]?
 local function setup_env(env, clear_env)
-  if clear_env then
-    return env
+  if not clear_env then
```

Comment:
In the old code if `env` was `nil` and `clear_env` was `true`, this would return `nil`, and this is what Gitsigns was relying on.